mirascope/mistral/types.py

```diff
@@ -0,0 +1,208 @@
+
+    response = Bookrecommender(genre="fantasy").call()
+    print(response.content)
+    #> The Name of the Wind
+
+    print(response.message)
+    #> ChatMessage(content='The Name of the Wind', role='assistant',
+    #  function_call=None, tool_calls=None)
+
+    print(response.choices)
+    #> [Choice(finish_reason='stop', index=0, logprobs=None,
+    #  message=ChatMessage(content='The Name of the Wind', role='assistant',
+    #  function_call=None, tool_calls=None))]
+    ```
+
+    """
+
+    @property
+    def choices(self) -> list[ChatCompletionResponseChoice]:
+        """Returns the array of chat completion choices."""
+        return self.response.choices
+
+    @property
+    def choice(self) -> ChatCompletionResponseChoice:
+        """Returns the 0th choice."""
+        return self.choices[0]
+
+    @property
+    def message(self) -> ChatMessage:
+        """Returns the message of the chat completion for the 0th choice."""
+        return self.choice.message
+
+    @property
+    def content(self) -> str:
+        """The content of the chat completion for the 0th choice."""
+        if isinstance(self.message.content, str):
+            return self.message.content
+        else:
+            return self.message.content[0]
+
+    @property
+    def tool_calls(self) -> Optional[list[ToolCall]]:
+        """Returns the tool calls for the 0th choice message."""
+        return self.message.tool_calls
+
+    @property
+    def tools(self) -> Optional[list[MistralTool]]:
+        """Returns the tools for the 0th choice message.
+
+        Raises:
+            ValidationError: if the tool call doesn't match the tool's schema.
+        """
+        if not self.tool_types or not self.tool_calls or len(self.tool_calls) == 0:
+            return None
+
+        extracted_tools = []
+        for tool_call in self.tool_calls:
+            for tool_type in self.tool_types:
+                if tool_call.function.name == tool_type.__name__:
+                    extracted_tools.append(tool_type.from_tool_call(tool_call))
+                    break
+
+        return extracted_tools
+
+    @property
+    def tool(self) -> Optional[MistralTool]:
+        """Returns the 0th tool for the 0th choice message.
+
+        Raises:
+            ValidationError: if the tool call doesn't match the tool's schema.
+        """
+        if not self.tool_types or not self.tool_calls or len(self.tool_calls) == 0:
+            return None
+
+        tool_call = self.tool_calls[0]
+        for tool_type in self.tool_types:
+            if self.tool_calls[0].function.name == tool_type.__name__:
+                return tool_type.from_tool_call(tool_call)
+
+        return None
+
+    def dump(self) -> dict[str, Any]:
+        """Dumps the chat completion to a dictionary."""
+        return {
+            "start_time": self.start_time,
+            "end_time": self.end_time,
+            "output": self.response.model_dump(),
+        }
+
+    def __str__(self):
```

Comment:
remove -- new interface doesn't override str method except for `BasePrompt` specifically for printing the prompt